client/src/grpc_client.rs

```diff
@@ -12,35 +12,35 @@ use admission_control_proto::{
 use anyhow::{bail, Result};
 use futures::Future;
 use grpcio::{CallOption, ChannelBuilder, EnvBuilder};
-use libra_crypto::ed25519::*;
 use libra_logger::prelude::*;
+use libra_types::crypto_proxies::EpochInfo;
 use libra_types::{
     access_path::AccessPath,
     account_address::AccountAddress,
     account_config::get_account_resource_or_default,
     account_state_blob::{AccountStateBlob, AccountStateWithProof},
     contract_event::{ContractEvent, EventWithProof},
-    crypto_proxies::ValidatorVerifier,
     get_with_proof::{
         RequestItem, ResponseItem, UpdateToLatestLedgerRequest, UpdateToLatestLedgerResponse,
     },
     transaction::{Transaction, Version},
     vm_error::StatusCode,
 };
 use std::convert::TryFrom;
-use std::sync::Arc;
+use std::sync::{Arc, Mutex};
 
 const MAX_GRPC_RETRY_COUNT: u64 = 1;
 
 /// Struct holding dependencies of client.
 pub struct GRPCClient {
     client: AdmissionControlClient,
-    validator_verifier: Arc<ValidatorVerifier>,
+    known_version: Arc<Mutex<Version>>,
+    epoch_info: Arc<Mutex<EpochInfo>>,
```

Comment:
Maybe rename to nighest_known_version?

Reply:
Also, given that known_version and epoch_info are always updated at the same time in LedgerInfo, there is a slight race condition that they can be out-of-sync if there are parallel calls to get_with_proof_async().  Could we just update them under the same Mutex?  Maybe put them in the same struct EpochInfoWithVersion?

Reply:
I think `known_version` is clear and it's used for a few places - those API refers to `client_known_version`

We don't have any concurrency running in the client but I can wrap them with one mutex.